enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,308 @@
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength:=80
+  // +kubebuilder:validation:Pattern:=[a-zA-Z0-9_-]
+  // +required
+  Name string `json:"name"`
+}
+
+// AzureCSIDriverConfigSpec defines properties that can be configured for the Azure CSI driver.
+type AzureCSIDriverConfigSpec struct{
+  // DiskEncryptionSet sets the cluster default storage class to encrypt volumes with a
+  // customer-managed encryption set, rather than the default platform-managed keys.
+  // +optional
+  DiskEncryptionSet *AzureDiskEncryptionSet `json:"diskEncryptionSet,omitempty"`
+}
+
+```
+
+#### GCP
+
+```go
+// GCPKMSKeyReference gathers required fields for looking up a GCP KMS Key
+type GCPKMSKeyReference struct {
+  // Name is the name of the customer-managed encryption key to be used for disk encryption.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:Pattern:=[a-zA-Z0-9_-]{1,63}
+  Name string `json:"name"`
+  
+  // KeyRing is the name of the KMS Key Ring which the KMS Key belongs to.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:Pattern:=[a-zA-Z0-9_-]{1,63}
+  KeyRing string `json:"keyRing"`
+  
+  // ProjectID is the ID of the Project in which the KMS Key Ring exists.
+  // Defaults to the cluster project defined in the cluster infrastructure object, if not set.
+  // +optional
+  ProjectID string `json:"projectID,omitempty"`
+  
+  // Location is the GCP location in which the Key Ring exists.
+  // Defaults to global, if not set.
+  // +optional
+  Location string `json:"location,omitempty"`
+}
+
+// GCPCSIDriverConfigSpec defines properties that can be configured for the GCP CSI driver.
+type GCPCSIDriverConfigSpec struct {
+    // KMSKey sets the cluster default storage class to encrypt volumes with customer-supplied
+    // encryption keys, rather than the default keys managed by GCP. 
+    // +optional
+    KMSKey *GCPKMSKeyReference `json:"kmsKey,omitempty"`
+}
+```
+
+#### Install Config
+
+The API extensions above correspond to types already in use in the Installer install config: 
+[AWS](https://github.com/openshift/installer/blob/master/pkg/types/aws/machinepool.go#L100), 
+[GCP](https://github.com/openshift/installer/blob/master/pkg/types/gcp/machinepools.go#L73), &
+[Azure](https://github.com/openshift/installer/blob/master/pkg/types/azure/disk.go#L36).
+
+### Implementation Details/Notes/Constraints
+
+The Installer will create a `ClusterCSIDriver` manifest when users have defined
+encryption keys on the `defaultMachinePlatform` (`installconfig.platform.(aws|azure|gcp).defaultMachinePlatform`).
+
+1. User adds encryption keys to the `defaultMachinePlatform` in the install config.
+2. Installer checks if `defaultMachinePlatform` has encryption keys defined, sets them in the `ClusterCSIDriver` manifest.
+3. `ClusterCSIDriver` object is created during install.
+4. cluster-storage-operator/CSI driver monitors `ClusterCSIDriver` and creates default storage class
+accordingly
```

Comment:
Will the installer create ClusterCSIDriver _only_ when an encryption key is defined?
It's [created by CSO](https://github.com/openshift/cluster-storage-operator/blob/master/assets/csidriveroperators/aws-ebs/standalone/10_cr.yaml) today, just trying to understand if that should still be the case when no encyption key is provided.

Reply:
> Will the installer create ClusterCSIDriver only when an encryption key is defined?

Good question. Yes, this was my intention, but I do not have strong opinions on it. Based on the earlier [comment](https://github.com/openshift/enhancements/pull/1163#discussion_r1064726391) from Jan, I believe this is ok. I will update the enhancement to be a bit more clear on this.